internal/provider/common/adcdebugserver.go

```diff
@@ -0,0 +1,341 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package common
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+
+	adctypes "github.com/apache/apisix-ingress-controller/api/adc"
+	"github.com/apache/apisix-ingress-controller/internal/adc/cache"
+	"github.com/apache/apisix-ingress-controller/internal/types"
+)
+
+type ResourceInfo struct {
+	ID   string
+	Name string
+	Type string
+	Link string
+}
+
+type ADCDebugProvider struct {
+	store         *cache.Store
+	configManager *ConfigManager[types.NamespacedNameKind, adctypes.Config]
+	pathPrefix    string
+}
+
+func newTemplate(name, body string) *template.Template {
+	return template.Must(template.New(name).
+		Funcs(template.FuncMap{"urlencode": url.QueryEscape}).
+		Parse(body))
+}
+
+func (asrv *ADCDebugProvider) SetupHandler(pathPrefix string, mux *http.ServeMux) {
+	asrv.pathPrefix = pathPrefix
+	mux.HandleFunc("/config", asrv.handleConfig)
+	mux.HandleFunc("/", asrv.handleIndex)
```

Comment:
Can those APIs ensure that the resources they output are managed by the current controller? For example, if an HTTPRoute is managed by another ingress controller, can it be listed by the current API?